sim-lib/src/sim_node.rs

```diff
@@ -1035,3 +1038,742 @@ impl UtxoLookup for UtxoValidator {
+                htlc_amount,
+                htlc_fee * 3
+            )
+            .is_ok());
+    }
+
+    /// Test addition of outgoing htlc to local state.
+    #[test]
+    fn test_check_outgoing_addition() {
+        // Create test channel with low local liquidity so that we run into failures.
+        let local_balance = 100_000;
+        let mut channel_state =
+            ChannelState::new(create_test_policy(local_balance / 2), local_balance);
+
+        let mut htlc = Htlc {
+            amount_msat: channel_state.policy.max_htlc_size_msat + 1,
+            cltv_expiry: channel_state.policy.cltv_expiry_delta,
+        };
+        // HTLC maximum size exceeded.
+        assert!(matches!(
+            channel_state.check_outgoing_addition(&htlc),
+            Err(ForwardingError::MoreThanMaximum(_, _))
+        ));
+
+        // Beneath HTLC minimum size.
+        htlc.amount_msat = channel_state.policy.min_htlc_size_msat - 1;
+        assert!(matches!(
+            channel_state.check_outgoing_addition(&htlc),
+            Err(ForwardingError::LessThanMinimum(_, _))
+        ));
+
+        // Add two large htlcs so that we will start to run into our in-flight total amount limit.
+        let hash_1 = PaymentHash { 0: [1; 32] };
+        let htlc_1 = Htlc {
+            amount_msat: channel_state.policy.max_in_flight_msat / 2,
+            cltv_expiry: channel_state.policy.cltv_expiry_delta,
+        };
+
+        assert!(channel_state.check_outgoing_addition(&htlc_1).is_ok());
+        assert!(channel_state.add_outgoing_htlc(hash_1, htlc_1).is_ok());
+
+        let hash_2 = PaymentHash { 0: [2; 32] };
+        let htlc_2 = Htlc {
+            amount_msat: channel_state.policy.max_in_flight_msat / 2,
+            cltv_expiry: channel_state.policy.cltv_expiry_delta,
+        };
+
+        assert!(channel_state.check_outgoing_addition(&htlc_2).is_ok());
+        assert!(channel_state.add_outgoing_htlc(hash_2, htlc_2).is_ok());
+
+        // Now, assert that we can't add even our smallest htlc size, because we're hit our in-flight amount limit.
+        htlc.amount_msat = channel_state.policy.min_htlc_size_msat;
+        assert!(matches!(
+            channel_state.check_outgoing_addition(&htlc),
+            Err(ForwardingError::ExceedsInFlightTotal(_, _))
+        ));
+
+        // Resolve both of the htlcs successfully so that the local liquidity is no longer available.
+        assert!(channel_state.remove_outgoing_htlc(&hash_1).is_ok());
+        channel_state.settle_outgoing_htlc(htlc_1.amount_msat, true);
+
+        assert!(channel_state.remove_outgoing_htlc(&hash_2).is_ok());
+        channel_state.settle_outgoing_htlc(htlc_2.amount_msat, true);
+
+        // Now we're going to add many htlcs so that we hit our in-flight count limit (unique payment hash per htlc).
+        for i in 0..channel_state.policy.max_htlc_count {
+            let hash = PaymentHash {
+                0: [i.try_into().unwrap(); 32],
```

Comment:
This test might become flaky due to the conversion of a `u64` value to a `u8`. If `max_htlc_count` exceeds 255, this could lead to a panic.

Reply:
> become flaky 

It'll just outright fail rather than flake, so I think we're okay ot leave as-is for now. If we add tests that hit that count, then we can update this? Would rather just keep it simple to start (don't see a need to test > 255 IMO)